.dotnet/src/Custom/Chat/ChatMessageContent.Serialization.cs

```diff
@@ -0,0 +1,50 @@
+﻿using System;
+using System.ClientModel.Primitives;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Linq;
+using System.Text.Json;
+
+namespace OpenAI.Chat;
+
+public partial class ChatMessageContent
+{
+    internal void WriteTo(Utf8JsonWriter writer, ModelReaderWriterOptions options = null)
+    {
+        if (Count == 0)
+        {
+            writer.WriteNullValue();
+        }
+        else if (Count == 1 && this[0].Kind == ChatMessageContentPartKind.Text)
```

Comment:
nit: In the serialization methods of the concrete ChatMessage types (e.g., `UserChatMessage.Serialization.cs`), this check is done as:

```csharp
if (Content.Count == 1 && Content[0].Text != null)
```

I think checking for "Kind == Text" is technically more correct. Could you make it the same across the other ChatMessage types for consistency?

Reply:
Ah, good note -- the motivation for putting those helpers with `ChatMessageContent` directly was to facilitate standardization here, so I'll go a step further and actually make each of those concrete message types call `Content.WriteTo(writer, options)` after checking the inner collection being defined.